src/runtime_src/xrt/device/hal2.cpp

```diff
@@ -216,7 +216,13 @@ std::shared_ptr<xrt_core::device>
 device::
 get_core_device() const
 {
-  return m_handle.get_handle();
+  if (m_handle) {
+    if (auto cdev = m_handle.get_handle())
+      return cdev;
+  }
```

Comment:
```suggestion
    if (auto cdev = m_handle.get_handle())
      return cdev;
```
m_handle is an xrt::device, you can safely get the pimpl (get_handle()) from it, but the pimpl can be nullptr if the device object is not initialized (opened).